### PR TITLE
[Merged by Bors] - feat(Algebra/Star/Order): more lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Star/Order.lean [hunk 1]
```diff
@@ -185,4 +185,85 @@ theorem conjugate_le_conjugate' {a b : R} (hab : a ≤ b) (c : R) : c * a * star
...
+  rw [StarOrderedRing.le_iff] at h
+  obtain ⟨d, hd, rfl⟩ := h
+  rw [star_add, hx]
+  induction hd using AddSubmonoid.closure_induction' generalizing x with
+  | Hs _ h => obtain ⟨x, rfl⟩ := h; simp
+  | H1 => simp
+  | Hmul a _ b _ ha hb =>
+    replace ha : star a = a := by simpa using ha (star_zero _)
+    replace hb : star b = b := by simpa using hb (star_zero _)
+    simp only [star_add, ha, hb]
```
Comment (j-loreaux) [inline Mathlib/Algebra/Star/Order.lean:243]: Again, a bit more conceptual using the fact that `star` is the identity on the range of `s ↦ star s * s`:
```suggestion
  rw [StarOrderedRing.le_iff] at h
  obtain ⟨d, hd, rfl⟩ := h
  rw [IsSelfAdjoint, star_add, hx.star_eq]
  congr
  refine AddMonoidHom.eqOn_closureM (f := starAddEquiv (R := R)) (g := .id R) ?_ hd
  rintro - ⟨s, rfl⟩
  simp
```

**Review by j-loreaux — COMMENTED** [review]
I changed the induction proofs. Looks good.

**Comment by j-loreaux** [comment]
bors d+

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**